fix: replace std::time with instant for wasm

gudnuf opened the pull request:
### Description

`std::time` is unusable in wasm environments. I replaced all usage of `std::time` relevant to a wasm wallet with the `instant` crate. For native targets instant is just a type alias for std::time

-----

### Notes to the reviewers

Side note: the [instant](https://crates.io/crates/instant) documentation says its no longer maintained, but cdk already used instant so I decided to keep it. 

-----

### Suggested [CHANGELOG](https://github.com/cashubtc/cdk/blob/main/CHANGELOG.md) Updates

<!-- Please do not edit the actual changelog but note what you changed here. -->

#### CHANGED

#### ADDED

#### REMOVED

#### FIXED

----

### Checklist

* [x] I followed the [code style guidelines](https://github.com/cashubtc/cdk/blob/main/CODE_STYLE.md)
* [x] I ran `just final-check` before committing

thesimplekid commented:
Since you pointed out instant is not maintained can we remove it? Seems like it should be pretty easy to just switch out for https://docs.rs/web-time/latest/web_time/

gudnuf commented:
> Since you pointed out instant is not maintained can we remove it? Seems like it should be pretty easy to just switch out for https://docs.rs/web-time/latest/web_time/

I removed instant and replaced with web_time, but I left the rest of the usage of std::time that doesn't seem to be used by a wasm wallet
